File: api/app/schemas/image_processing.py
"""Pydantic-Schemas für Bild-Nachbearbeitung (Crop, Resize, Center, Pad-Square)."""

from pydantic import BaseModel, Field


class CropRequest(BaseModel):
    """Request für POST /assets/{asset_id}/image/crop."""

    source_file: str = Field(..., description="z.B. image_bgremoved.png")
    x: int = Field(..., ge=0, description="X-Koordinate links oben")
    y: int = Field(..., ge=0, description="Y-Koordinate links oben")
    width: int = Field(..., gt=0, description="Crop-Breite in Pixel")
    height: int = Field(..., gt=0, description="Crop-Höhe in Pixel")


class ResizeRequest(BaseModel):
    """Request für POST /assets/{asset_id}/image/resize."""

    source_file: str = Field(..., description="z.B. image_bgremoved.png")
    width: int = Field(..., gt=0, description="Ziel-Breite")
    height: int = Field(..., gt=0, description="Ziel-Höhe")
    maintain_aspect: bool = Field(
        default=True,
        description="Seitenverhältnis beibehalten (mit Padding)",
    )


class CenterRequest(BaseModel):
    """Request für POST /assets/{asset_id}/image/center."""

    source_file: str = Field(..., description="z.B. image_bgremoved.png")
    padding: float = Field(
        default=0.1,
        ge=0.0,
        le=1.0,
        description="Padding als Anteil der Subjekt-Größe (0.1 = 10%)",
    )


class PadSquareRequest(BaseModel):
    """Request für POST /assets/{asset_id}/image/pad-square."""

    source_file: str = Field(..., description="z.B. image_bgremoved.png")
    background: str = Field(
        default="white",
        description="white | black | transparent",
    )


class ImageProcessingResponse(BaseModel):
    """Response für alle Bild-Processing-Endpunkte."""

    output_file: str
    width: int
    height: int
    file_size_bytes: int

File: frontend/src/api/imageProcessing.ts
import { apiClient } from "./client.js";

export interface ImageProcessingResponse {
  output_file: string;
  width: number;
  height: number;
  file_size_bytes: number;
}

export async function getImageSources(
  assetId: string
): Promise<{ sources: string[] }> {
  const { data } = await apiClient.get<{ sources: string[] }>(
    `/assets/${assetId}/image/sources`
  );
  return data;
}

export async function cropImage(
  assetId: string,
  req: {
    source_file: string;
    x: number;
    y: number;
    width: number;
    height: number;
  }
): Promise<ImageProcessingResponse> {
  const { data } = await apiClient.post<ImageProcessingResponse>(
    `/assets/${assetId}/image/crop`,
    req
  );
  return data;
}

export async function resizeImage(
  assetId: string,
  req: {
    source_file: string;
    width: number;
    height: number;
    maintain_aspect?: boolean;
  }
): Promise<ImageProcessingResponse> {
  const { data } = await apiClient.post<ImageProcessingResponse>(
    `/assets/${assetId}/image/resize`,
    req
  );
  return data;
}

export async function centerImage(
  assetId: string,
  req: {
    source_file: string;
    padding?: number;
  }
): Promise<ImageProcessingResponse> {
  const { data } = await apiClient.post<ImageProcessingResponse>(
    `/assets/${assetId}/image/center`,
    req
  );
  return data;
}

export async function padSquareImage(
  assetId: string,
  req: {
    source_file: string;
    background?: string;
  }
): Promise<ImageProcessingResponse> {
  const { data } = await apiClient.post<ImageProcessingResponse>(
    `/assets/${assetId}/image/pad-square`,
    req
  );
  return data;
}
